[package] icu/67.1: can't compile on Windows

<!-- 
  Please don't forget to update the issue title.
  Include all applicable information to help us reproduce your problem.
-->

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **icu/67.1**
  * Operating System+version: **Windows 10**
  * Compiler+version: **MSVC 2019**
  * Conan version: **conan 1.28.1**


### Conan profile
```
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=14
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
*:fPIC=True
*:shared=False
*:static=True
[build_requires]
[env]
```


### Steps to reproduce (Include if Applicable)
conan install icu/67.1@ --build "*"


### Logs (Include/Attach if Applicable)
<details><summary>Click to expand log</summary>

```
Configuration:
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.cppstd=14
compiler.runtime=MD
compiler.version=16
os=Windows
os_build=Windows
[options]
*:fPIC=True
*:shared=False
*:static=True
[build_requires]
[env]

icu/67.1: Forced build from source
Installing package: icu/67.1
Requirements
    icu/67.1 from 'artifactory' - Cache
Packages
    icu/67.1:1227fdf5511e684ad7c925afa05d82715a7d91b5 - Build

Installing (downloading, building) binaries...
icu/67.1: WARN: Build folder is dirty, removing it: D:\.conan\.data\icu\67.1\_\_\build\1227fdf5511e684ad7c925afa05d82715a7d91b5
icu/67.1: Copying sources to build folder
icu/67.1: Building your package in D:\.conan\.data\icu\67.1\_\_\build\1227fdf5511e684ad7c925afa05d82715a7d91b5
icu/67.1: Generator txt created conanbuildinfo.txt
icu/67.1: Calling build()
**********************************************************************
** Visual Studio 2019 Developer Command Prompt v16.7.1
** Copyright (c) 2020 Microsoft Corporation
**********************************************************************
[vcvarsall.bat] Environment initialized for: 'x64'
icu/67.1: run_in_windows_bash: "C:\Program Files\Git\bin\bash.exe" --login -c ^"cd \^"/d/.conan/.data/icu/67.1/_/_/build/1227fdf5511e684ad7c925afa05d82715a7d91b5/source_subfolder/build\^" ^&^& ../source/runConfigureICU Cygwin/MSVC --prefix=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5 --with-library-bits=64 --disable-samples --disable-layout --disable-layoutex --disable-extras --with-data-packaging=archive --datarootdir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/lib --sbindir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/bin --enable-static --disable-shared --disable-tests ^"
icu/67.1: export CPP= CC=cl CXX=cl CPPFLAGS=-DNDEBUG -DU_STATIC_IMPLEMENTATION CFLAGS=-Gy -MD -O2 -Ob2 -MD CXXFLAGS=-Gy -MD -O2 -Ob2 -MD /std:c++14 LDFLAGS= MAKE=
icu/67.1: Running ./configure  --prefix=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5 --with-library-bits=64 --disable-samples --disable-layout --disable-layoutex --disable-extras --with-data-packaging=archive --datarootdir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/lib --sbindir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/bin --enable-static --disable-shared --disable-tests for Windows with Cygwin using Microsoft Visual C++ compiler
icu/67.1:
icu/67.1: checking for ICU version numbers... release 67.1, library 67.1, unicode version 13.0
icu/67.1: checking build system type... x86_64-pc-mingw64
icu/67.1: checking host system type... x86_64-pc-mingw64
icu/67.1: checking whether to build debug libraries... no
icu/67.1: checking whether to build release libraries... yes
icu/67.1: checking whether the C compiler works... yes
icu/67.1: checking for C compiler default output file name... conftest.exe
icu/67.1: checking for suffix of executables... .exe
icu/67.1: checking whether we are cross compiling... no
icu/67.1: checking for suffix of object files... obj
icu/67.1: checking whether we are using the GNU C compiler... no
icu/67.1: checking whether cl accepts -g... yes
icu/67.1: checking for cl option to accept ISO C89... none needed
icu/67.1: checking whether we are using the GNU C++ compiler... no
icu/67.1: checking whether cl accepts -g... yes
icu/67.1: checking for pkg-config... no
icu/67.1: checking for ICULEHB... no
icu/67.1: checking how to run the C preprocessor... cl -E
icu/67.1: checking for a BSD-compatible install... /usr/bin/install -c
icu/67.1: checking for python3... python3
icu/67.1: checking for gmake... no
icu/67.1: checking for gnumake... no
icu/67.1: checking for doxygen... no
icu/67.1: configure: error: link.exe is not a valid linker. Your PATH is incorrect.
icu/67.1:                   Please follow the directions in ICU's readme.
icu/67.1: ../source/runConfigureICU: ./configure failed
icu/67.1:
icu/67.1: ERROR: Package '1227fdf5511e684ad7c925afa05d82715a7d91b5' build failed
icu/67.1: WARN: Build folder D:\.conan\.data\icu\67.1\_\_\build\1227fdf5511e684ad7c925afa05d82715a7d91b5
ERROR: icu/67.1: Error in build() method, line 94
        self.run(self._build_config_cmd, win_bash=tools.os_info.is_windows)
        ConanException: Error 1 while executing ../source/runConfigureICU Cygwin/MSVC --prefix=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5 --with-library-bits=64 --disable-samples --disable-layout --disable-layoutex --disable-extras --with-data-packaging=archive --datarootdir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/lib --sbindir=D:/.conan/.data/icu/67.1/_/_/package/1227fdf5511e684ad7c925afa05d82715a7d91b5/bin --enable-static --disable-shared --disable-tests
```

</details>

--- a/recipes/icu/all/conanfile.py
+++ b/recipes/icu/all/conanfile.py
@@ -60,8 +60,7 @@ class ICUBase(ConanFile):
         del self.info.options.silent  # Verbosity doesn't affect package's ID
 
     def build_requirements(self):
-        if tools.os_info.is_windows and "CONAN_BASH_PATH" not in os.environ and \
-                tools.os_info.detect_windows_subsystem() != "msys2":
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
             self.build_requires("msys2/20200517")
 
     def source(self):

